src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/contactperson/ContactPersonMigrationIntTest.kt

```diff
@@ -283,6 +303,26 @@ class ContactPersonMigrationIntTest : SqsIntegrationTestBase() {
         assertThat(request.modifyUsername).isEqualTo("ADJUA.MENSAH")
         assertThat(request.modifyDateTime).isEqualTo(LocalDateTime.parse("2024-01-02T10:23"))
       }
+
+      @Test
+      fun `will send mandatory core person data to DPS`() {
+        val request = requests.find { it.personId == 2000L } ?: throw AssertionError("Request not found")
+        assertThat(request.personId).isEqualTo(2000L)
+        assertThat(request.firstName).isEqualTo("KWAME")
+        assertThat(request.lastName).isEqualTo("KOBE")
+        assertThat(request.dateOfBirth).isNull()
+        assertThat(request.gender).isNull()
+        assertThat(request.title).isNull()
+        assertThat(request.language).isNull()
+        assertThat(request.interpreterRequired).isFalse()
+        assertThat(request.domesticStatus).isNull()
+        assertThat(request.deceasedDate).isNull()
+        assertThat(request.staff).isFalse()
+        assertThat(request.createUsername).isEqualTo("ADJUA.BEEK")
+        assertThat(request.createDateTime).isEqualTo(LocalDateTime.parse("2022-01-02T10:23"))
+        assertThat(request.modifyUsername).isNull()
+        assertThat(request.modifyDateTime).isNull()
```

Comment:
You could lose some nosie with:
```
with(requests.find { it.personId == 2000L }!!) {
  assertThat(firstName).isEqualTo("KWAME")
  ...etc
}
```
You lose the reference to `request` everywhere and the !! would fail the test if request not found.

Reply:
Or you could continue to throw for the nice error message 🤷